### PR TITLE
CCS-3891-metadata_bugfix

### PR DESCRIPTION
**Pull request by xdavidson**
this PR made the following changes:
  * fix showMetadataAlertIcon issue
  * fix metadata population on draft version

### CODE REVIEW

File: pantheon-bundle/frontend/src/app/versions.test.tsx [hunk 1]
```diff
@@ -210,7 +210,7 @@ describe("Versions tests", () => {
         const instance = wrapper.instance()
         wrapper.setState({ "login": true })
         wrapper.setState({
-            "results": [[{ "type": "draft", "icon": "BlankImage", "path": "/modules/test", "version": "Version 1", "publishedState": "Not published", "updatedDate": "", "firstButtonType": "primary", "secondButtonType": "secondary", "firstButtonText": "Publish", "secondButtonText": "Preview", "isDropdownOpen": false, "isArchiveDropDownOpen": false, "metadata": "" }]],
```
Comment (rednitish) [inline pantheon-bundle/frontend/src/app/versions.test.tsx:213]: should this be changed ?,here were we testing , if states get changed on publich event from draft to release ?

File: pantheon-bundle/frontend/src/app/versions.tsx [hunk 2]
```diff
@@ -34,6 +34,14 @@ export interface IProps {
     onPublishEvent: () => void
 }
 
+// Define properties in Metadata
+export interface IMetadata {
```
Comment (rednitish) [inline pantheon-bundle/frontend/src/app/versions.tsx:38]: +1:  neat

File: pantheon-bundle/frontend/src/app/versions.test.tsx (hunk 1 above)
Comment (xdavidson) [reply pantheon-bundle/frontend/src/app/versions.test.tsx:213]: changePublishState handles:
  * change draft to released
  or
  * change released to draft

On line 104 there is a test for changePublishState that matches "Publish" button. That's why I changed this test to cover the other scenario,

**Comment by xdavidson** [comment]
test the bot notification setting

File: pantheon-bundle/frontend/src/app/versions.test.tsx (hunk 1 above)
Comment (xdavidson) [reply pantheon-bundle/frontend/src/app/versions.test.tsx:213]: I updated the tests so that it covers the click for both scenarios!

**Review by rednitish — APPROVED** [review]
Thanks for the changes, changes LGTM

**Pull request merged**